tools/caffe.cpp

```diff
@@ -34,6 +34,13 @@ DEFINE_string(solver, "",
     "The solver definition protocol buffer text file.");
 DEFINE_string(model, "",
     "The model definition protocol buffer text file.");
+DEFINE_string(phase, "",
+    "Optional; network phase (TRAIN or TEST). Only used for 'time'.");
+DEFINE_int32(level, 0,
```

Comment:
Would we just set the `train_state` and not override any of the `test_states`?
https://github.com/BVLC/caffe/blob/rc3/src/caffe/proto/caffe.proto#L135-L136